django/db/models/aggregates.py

```diff
@@ -120,16 +123,6 @@ def resolve_expression(
     ):
         # Aggregates are not allowed in UPDATE queries, so ignore for_save
         c = super().resolve_expression(query, allow_joins, reuse, summarize)
-        c.filter = (
-            c.filter.resolve_expression(query, allow_joins, reuse, summarize)
-            if c.filter
-            else None
-        )
-        c.order_by = (
-            c.order_by.resolve_expression(query, allow_joins, reuse, summarize)
-            if c.order_by
-            else None
-        )
```

Comment:
I think this was missed in the multiple iterations of the patch but the goal of moving `filter` and `order_by` to `source_expressions` was to avoid these two lines. It makes sense that the double resolving (from `super()`) breaks `OuterRef`.